scripts/firedrake-zenodo

```diff
@@ -0,0 +1,252 @@
+                   help="Output to the named bibtex file rather than firedrake-zenodo.bib")
+parser.add_argument("--message", "-m", action="store", nargs=1,
+                    help="Short description of the reason for this release. E.g. Version of Firedrake used in <paper name>.")
+for component in components:
+    parser.add_argument("--%s" % component, action="store", nargs=1,
+                        help="Use this git hash for %s instead of that in the file or the checked out version."
+                        % component)
+
+parser.add_argument("--log", action='store_true',
+                    help="Produce a verbose log of the installation process in firedrake-zenodo.log. If you have problem running this script, please include this log in any bug report you file.")
+
+
+args = parser.parse_args()
+
+cwd = os.getcwd()
+src = os.path.dirname(__file__) + "/../src"
+
+# Set up logging
+if args.log:
+    # Log to file at DEBUG level
+    logging.basicConfig(level=logging.DEBUG,
+                        format='%(asctime)s %(levelname)-6s %(message)s',
+                        filename='firedrake-zenodo.log',
+                        filemode='w')
+    # Log to console at INFO level
+    console = logging.StreamHandler()
+    console.setLevel(logging.INFO)
+    formatter = logging.Formatter('%(message)s')
+    console.setFormatter(formatter)
+    logging.getLogger().addHandler(console)
+else:
+    # Log to console at INFO level
+    logging.basicConfig(level=logging.INFO,
+                        format='%(message)s')
+log = logging.getLogger()
+
+
+def check_call(arguments):
+    if args.log:
+        try:
+            log.debug(subprocess.check_output(arguments, stderr=subprocess.STDOUT))
+        except subprocess.CalledProcessError as e:
+            log.debug(e.output)
+            raise
+    else:
+        subprocess.check_call(arguments)
+
+
+def check_output(args):
+    try:
+        return subprocess.check_output(args, stderr=subprocess.STDOUT)
+    except subprocess.CalledProcessError as e:
+        log.debug(e.output)
+        raise
+
+
+def collect_repo_shas():
+    shas = {}
+
+    for component in components:
+        try:
+            os.chdir(src + "/" + component)
+            try:
+                check_call(["git", "diff-index", "--quiet", "HEAD"])
+            except subprocess.CalledProcessError:
+                log.error("Component %s has uncommitted changes, cannot create release" % component)
+                sys.exit(0)
+            shas[component] = check_output(["git", "rev-parse", "HEAD"]).strip()
+        except:
+            log.error("Failed to retrieve git hash for %s" % component)
+            raise
+    return shas
+
+if args.release_tag:
+    tag = args.release_tag[0]
+    log.info("Retrieving BibTeX data for Firedrake release %s." % tag)
+    log.info("This may take a few seconds.")
+    try:
+        request = urlopen("http://firedrakeproject.org/zenodo-bibtex.cgi?tag=%s" % tag)
```

Comment:
This makes a web request to `firedrakeproject.org/zenodo-bibtex.cgi` passing a user-specified tag.  It is not sanitised here, so let's check that the cgi script does sanitising.